app/models/renewal/WhichCurrencies.scala

```diff
@@ -16,231 +16,139 @@
 
 package models.renewal
 
-import cats.data.Validated.{Invalid, Valid}
-import jto.validation.GenericRules._
 import jto.validation._
 import jto.validation.forms.UrlFormEncoded
-import models.FormTypes._
-import models._
-import models.moneyservicebusiness._
+import models.{currencies => currenciesSeq, moneyservicebusiness => msb}
 import play.api.libs.json._
 import utils.MappingUtils.Implicits._
+import utils.MappingUtils.constant
 import utils.{GenericValidators, TraversableValidators}
 
 case class WhichCurrencies(currencies: Seq[String],
-                           usesForeignCurrencies: Option[Boolean],
-                           bankMoneySource: Option[BankMoneySource],
-                           wholesalerMoneySource: Option[WholesalerMoneySource],
-                           customerMoneySource: Option[Boolean])
+                           usesForeignCurrencies: Option[UsesForeignCurrencies] = None,
+                           moneySources: Option[MoneySources] = None) {
 
+  def currencies(p: Seq[String]): WhichCurrencies =
+    this.copy(currencies = p)
 
-object WhichCurrencies {
+  def usesForeignCurrencies(p: UsesForeignCurrencies): WhichCurrencies =
+    this.copy(usesForeignCurrencies = Some(p))
+
+  def moneySources(p: MoneySources): WhichCurrencies =
+    this.copy(moneySources = Some(p))
+}
 
-  type MoneySourceValidation = (Option[BankMoneySource], Option[WholesalerMoneySource], Option[Boolean])
-  type WhichCurrenciesValidation = (Option[Boolean], Option[BankMoneySource], Option[WholesalerMoneySource], Option[Boolean])
+object WhichCurrencies {
 
-  private val emptyToNone: String => Option[String] = { x =>
```

Comment:
was there a reason to make the function object public, when it was private?

Reply:
Not really but can you explain how making it private would change anything please?

Reply:
we can have a session outside this PR about the impact of private/public on programs design.

Reply:
Private means it's only accessible within this class. Taking away private implies it can be called from outside of the class by entities that instantiate this class. Its commonly used incorrectly. I would suggest if the method was private before and it does not need to be made public then keep it private. What is the justification for removing the private keyword? Does something fail if this val is private?

Reply:
Nothing fails I know the difference between public and private that wasn't my question. I was simply asking how it would affect this specific function. 

The method can be called outside of this class but it isn't, does not having the private keyword affect compilation times or something? I feel as though this keyword would only matter if someone tried to call this method outside this class which is most likely not going to happen as it's simply pulling a value from an option. 

I am happy to change this I am simply curious as to why its needed.